pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -262,13 +271,14 @@ contract MultiRewards is IMultiRewards, IDistributor, ReentrancyGuard, MultiRewa
         IERC20 pool,
         uint256 amount,
         uint256 deadline,
+        address account,
         address recipient,
         uint8 v,
         bytes32 r,
         bytes32 s
-    ) external {
-        IERC20Permit(address(pool)).permit(msg.sender, address(this), amount, deadline, v, r, s);
-        stakeFor(pool, amount, recipient);
+    ) external nonReentrant {
+        IERC20Permit(address(pool)).permit(account, address(this), amount, deadline, v, r, s);
```

Comment:
This is dangerous. I can pull this signature out of a transaction in the mempool and then include it in another transaction which sets me as the recipient - stealing the user's BPT.

We'd need to check another signature which includes the recipient field as well to make this secure.

Reply:
Either that or enforce that `account == recipient`